### PR TITLE
Feature/login

### LINKED ISSUE #4: Login
When first arriving at the site, a user should be able to log in with a username and password.

As a customer:

- I should be able to login
- I will see a login page when I first visit the site
- I can log in with the following credentials:

username: customer50 (where 50 is the ID of the user)
password: overlook2021

- Upon successfully logging in, I should see my dashboard.

### PR DESCRIPTION
**Pull request by qrispi**
Files Changed: index.html, scripts.js, styles.css

Change Summary: This merge adds a login page with error handling to the site at initial load. Once valid input is received, the appropriate data is fetched and displayed. User can also now log out by clicking button in nav bar. Basic styling was added to login view, mainly for organizational purposes.

Functions:

- validateLogin: This function checks the input from the login form. Unique error messages are generated and displayed to the user if they fail to enter any input or if the input does not match required parameters. Only a username in the format of 'customer[number between 0 and 51]' and a password of 'overlook2021' will generate a successful response. Once input is valid, a welcome message asking the user to wait while data is gathered appears, the function that fetches all data is fired with the id from the login and then once data is collected and organized, clearLogin is fired.
- clearLogin: This function clears the input fields (in case log out is changed to not force a page refresh), hides the login page and calls the function to show the user data.
- logoutButton: This nav bar button has an event listener with inline anonymous handler that refreshes the page upon clicking it. This may be refactored later but currently is the best method to ensure that the data will be reset and ready to change users.

Closes #4

### CODE REVIEW

**Comment by qrispi** [comment]
Hi Kass, please take a look at my login page merge and let me know what you think!

@zertroz @robbiejaeger

File: src/scripts.js [hunk 1]
```diff
@@ -15,32 +15,68 @@ const searchRoomsButton = document.getElementById('searchRoomsButton');
...
 clearFiltersButton.addEventListener('click', clearRoomOptions);
 myReservationsButton.addEventListener('click', displayUserReservations);
+loginButton.addEventListener('click', validateLogin);
+loginForm.addEventListener('keyup', () => {if(event.key === 'Enter') validateLogin()});
+logoutButton.addEventListener('click', () => location.reload());
 filteredRooms.addEventListener('click', generateModal);
 modalBg.addEventListener('click', collapseModal);
 
-getData();
+function validateLogin() {
```
Comment (Zertroz) [inline src/scripts.js:38]: This whole function looks really solid. I tried something similar with the error handling, but couldn't get it to work. I'll follow along with what you did here, see where I went wrong.

File: src/scripts.js [hunk 2]
```diff
@@ -15,32 +15,68 @@ const searchRoomsButton = document.getElementById('searchRoomsButton');
...
         hotel = new Hotel(data[0].rooms);
         hotel.updateBookings(data[1].bookings);
         user = new Customer(data[2]);
     })
     .then(logData)
-    .then(displayUserReservations);
+    .then(clearLogin);
+}
+
+function clearLogin() {
```
Comment (Zertroz) [inline src/scripts.js:75]: Smart! Makes it easier to use and follow

File: dist/index.html [hunk 3]
```diff
@@ -16,9 +16,26 @@
...
+        <h3>Please Login to make or view a reservation!</h3>
+        <form id="loginForm">
+          <label for="usernameInput">Username:
+            <input type="text" id="usernameInput" required>
+          </label>
+          <label for="passwordInput">Password:
+            <input type="password" id="passwordInput" required>
+          </label>
+          <button id="loginButton">Login</button>
+          <h3 id="errorMessage"></h3>
```
Comment (Zertroz) [inline dist/index.html:35]: Simple, but works.

**Review by Zertroz — APPROVED** [review]

**Pull request merged**